<launch>
    <arg name="exposure" default="50" />
    <arg name="gain" default="100" />
    <arg name="FPS" default="20" />
    <arg name="imu_rate" default="200" />    
    <arg name="resolution_width" default="640" />
    <arg name="resolution_height" default="480" />
    <arg name="publish_raw" default="1" />

    <arg name="auto_exposure" default="true" />
    <arg name="auto_exposure_target_brightness" default="80" />
    <arg name="auto_exposure_controller_gain" default="0.05" />
    <arg name="auto_exposure_recompute_frequency" default="5" />
    <arg name="auto_exposure_transition_frequency" default="10" />
    <arg name="auto_exposure_max_step_size" default="2" />


    <node pkg="duo3d_ros" name="duo_node" type="duo_node" required="true" output="screen">
		<!-- Supported Device Names: [Mini, Mini-lx] -->
		<param name="device_name" 		value="Mini-lx" />
		<param name="device_serial_number" 	value="N-A" />

    	<param name="frame_id"   value="duo3d_camera" />
		<!-- 	Choose anywhere from 0.1-3000 Frames Per Second (FPS) for the DUO to publish at. Please only
		use up to one decimal place ONLY, if you need to. This does not require a specific rate, like 
		the duo_dense3d.launch file requires. 

		 56 FPS @ 752x480 ->  49
		 62 FPS @ 640x480
		123 FPS @ 640x240
		240 FPS @ 640x120
		 93 FPS @ 320x480
		184 FPS @ 320x240 -> 171
		360 FPS @ 320x120

		-->	
		<param name="FPS"   value="$(arg FPS)" />
		<param name="ImuRate"   value="$(arg imu_rate)" />

		<param name="resolution_width"	value="$(arg resolution_width)" />
		<param name="resolution_height"	value="$(arg resolution_height)" />
	    <param name="activate_undistort"   	value="true" />	    
	    <param name="activate_undistort"   	value="true" />	    

		<param name="exposure"      value="$(arg exposure)"  />
		<param name="gain"          value="$(arg gain)"  /> 
		<param name="led_lighting"	value="0"  />

	    <param name="use_DUO_imu"   	value="true" />
	    <param name="use_DUO_LEDs"   	value="true" />

	    <param name="VerticalFlip"   	value="false" />
	    <param name="HorizontalFlip"   	value="false" />
	    <param name="CameraSwap"     	value="true" />
	    
	    <param name="publish_raw" type="bool" value="$(arg publish_raw)" />

	    <!-- NOTE: to set the calibration manually you'll need to have the "ost.txt" file from the ros calibration. then you convert (use http://wiki.ros.org/camera_calibration_parsers) it to two separate files and put it into the calibration folder -->

	    <param name="camera_info_left" value="file://$(find duo3d_ros)/calibration/duo3d_left_640_480.yaml" />
	    <param name="camera_info_right" value="file://$(find duo3d_ros)/calibration/duo3d_right_640_480.yaml" />

		<param name="auto_exposure"   value="$(arg auto_exposure)" />
		<param name="auto_exposure_target_brightness"   value="$(arg auto_exposure_target_brightness)" />
		<param name="auto_exposure_controller_gain"   value="$(arg auto_exposure_controller_gain)" />
		<param name="auto_exposure_recompute_frequency"   value="$(arg auto_exposure_recompute_frequency)" />
		<param name="auto_exposure_transition_frequency"   value="$(arg auto_exposure_transition_frequency)" />
		<param name="auto_exposure_max_step_size"   value="$(arg auto_exposure_max_step_size)" />

<!--okvis ?-->
		<!--remap from="/duo3d_camera/left/image_raw" to="/cam0/image_raw" />
    	<remap from="/duo3d_camera/right/image_raw" to="/cam1/image_raw" />
    	<remap from="/calibration0" to="/cam0/calibration" />
    	<remap from="/calibration1" to="/cam1/calibration" />
    	<remap from="/duo3d_camera/cam_imu" to="/imu0" /-->

<!--rovio-->
		<!--remap from="/duo3d_camera/left/image_raw" to="/cam0/image_raw" />
    	<remap from="/duo3d_camera/right/image_raw" to="/cam1/image_raw" />
    	<remap from="/calibration0" to="/cam0/calibration" />
    	<remap from="/calibration1" to="/cam1/calibration" />
    	<remap from="/duo3d_camera/cam_imu" to="/imu0" /-->

<!--rcars-->
		<remap from="/duo3d_camera/left/image_raw" to="/cam0/image_raw" />
    	<remap from="/duo3d_camera/right/image_raw" to="/cam1/image_raw" />
    	<remap from="/calibration0" to="/cam0/calibration" />
    	<remap from="/calibration1" to="/cam1/calibration" />
    	<remap from="/duo3d_camera/cam_imu" to="/imu0" />

    </node>

</launch>
